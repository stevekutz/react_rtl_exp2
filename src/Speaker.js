import React, {Fragment} from 'react';


function Speaker(props) {
    const {speak, message = "" } = props;

    return (
        <Fragment>
            <button onClick = {speak}>Speak</button>
            <div>{message}</div>
        </Fragment>
    )

}

export default Speaker;