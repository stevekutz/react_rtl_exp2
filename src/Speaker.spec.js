import React from 'react';
import renderer from 'react-test-renderer';

import Speaker from './Speaker';
import {render, fireEvent} from '@testing-library/react';

import '@testing-library/react/cleanup-after-each';

describe(`<Speaker /> handles some props for testing !!`, () => {
    it('matches snapshot', () => {
        const tree = renderer.create(<Speaker/>);
    
        expect(tree.toJSON()).toMatchSnapshot();
    })

    it('matches snapshot with props', () => {
        const tree = renderer.create(<Speaker message = {'hello Prop'}/>);
    
        expect(tree.toJSON()).toMatchSnapshot();
    })

    it('matches snapshot with different props', () => {
        const tree = renderer.create(<Speaker message = {'hello AGAIN Prop'}/>);
    
        expect(tree.toJSON()).toMatchSnapshot();
    })

    it(`should call speak on button click, is onClick set up correctly`, () => {
        // our mock function,  !!
        // does not do anything BUT shows that a function was called
        const speak = jest.fn();
        
        const {getByText} = render(<Speaker speak = {speak}/>);

        const button = getByText(/Speak/);

        fireEvent.click(button);

        expect(speak).toHaveBeenCalled();
        expect(speak).toHaveBeenCalledTimes(1);
    })

    it('should display  message', () => {
        const {getByText} = render(<Speaker message = "something"/>);
        getByText(/something/);
        getByText("something");

    });
} )