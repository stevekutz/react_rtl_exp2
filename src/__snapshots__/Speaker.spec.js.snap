// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speaker /> handles some props for testing !! matches snapshot 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div>
    
  </div>,
]
`;

exports[`<Speaker /> handles some props for testing !! matches snapshot with different props 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div>
    hello AGAIN Prop
  </div>,
]
`;

exports[`<Speaker /> handles some props for testing !! matches snapshot with props 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div>
    hello Prop
  </div>,
]
`;
